tests/phpunit/api/v3/JobTest.php

```diff
@@ -330,6 +330,307 @@ public function testBatchMergeWorks($dataSet) {
   }
 
   /**
+   * Test the decisions made for addresses when merging.
+   *
+   * @dataProvider getMergeAddresses
+   *
+   * Scenarios:
+   * (the ones with **** could be disputed as whether it is the best outcome).
+   *   'matching_primary' - Primary matches, including location_type_id. One contact has an additional address.
+   *     - result - primary is the shared one. Additional address is retained.
+   *   'matching_primary_reverse' - Primary matches, including location_type_id. Keep both. (opposite order)
```

Comment:
Confirmed the bug for batch merging. Works OK when manual merging, so I guess addresses beyond the first one aren't getting 'selected' for migration for some reason. Posted on JIRA: https://issues.civicrm.org/jira/browse/CRM-19076